spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vectorstore/cassandra/CassandraVectorStoreAutoConfiguration.java

```diff
@@ -63,25 +62,21 @@ public CassandraVectorStore vectorStore(EmbeddingModel embeddingModel, Cassandra
 			ObjectProvider<VectorStoreObservationConvention> customObservationConvention,
 			BatchingStrategy batchingStrategy) {
 
-		var builder = CassandraVectorStoreConfig.builder().withCqlSession(cqlSession);
-
-		builder = builder.withKeyspaceName(properties.getKeyspace())
-			.withTableName(properties.getTable())
-			.withContentColumnName(properties.getContentColumnName())
-			.withEmbeddingColumnName(properties.getEmbeddingColumnName())
-			.withIndexName(properties.getIndexName())
-			.withFixedThreadPoolExecutorSize(properties.getFixedThreadPoolExecutorSize());
-
-		if (!properties.isInitializeSchema()) {
-			builder = builder.disallowSchemaChanges();
-		}
-		if (properties.getReturnEmbeddings()) {
-			builder = builder.returnEmbeddings();
-		}
-
-		return new CassandraVectorStore(builder.build(), embeddingModel,
-				observationRegistry.getIfUnique(() -> ObservationRegistry.NOOP),
-				customObservationConvention.getIfAvailable(() -> null), batchingStrategy);
+		return CassandraVectorStore.builder()
+			.session(cqlSession)
+			.keyspace(properties.getKeyspace())
+			.table(properties.getTable())
+			.contentColumnName(properties.getContentColumnName())
+			.embeddingColumnName(properties.getEmbeddingColumnName())
+			.indexName(properties.getIndexName())
+			.fixedThreadPoolExecutorSize(properties.getFixedThreadPoolExecutorSize())
+			.disallowSchemaChanges(!properties.isInitializeSchema())
+			.returnEmbeddings(properties.getReturnEmbeddings())
```

Comment:
there's a behaviour change here, in that we're no longer respecting the builder's default, imposing the autoConfiguration's default over it.  (the builder's default should be canonical)

i'm not sure how important this is… 🤷

Reply:
Good catch, we should be consistent.  I'll create a new issue to capture this and discuss.